_posts/2018-07-25-Relay-Networking-Deep-Dive.markdown

```diff
@@ -291,31 +346,47 @@ const batchRequestQuery = async (
 };
 ```
 
-`getDeferrableVariables` function will get variables from result data from other requests.
+## Relay Modern is very flexible
 
-`batchRequestQuery` function will execute each of the requests, and will `sink.next()` as soon as it has the GraphQL
-server response data. It will only close the Observable stream when all requests has been fullfiled.
+Depending on your application needs, you can scale from a simpler Promise-based API for your custom network layer to
+one that uses Observables to always resolves from cache data first and then resolves from the server.
 
---
+Here are some production examples:
 
-Relay Modern is very flexible!
+- [Artsy Emission][artsy]: Uses the Promise API, caches the results locally, and shares logic with native code in an
+  iOS app so that queries can be pre-cached before the JavaScript runtime has started.
 
-You can have a custom network layer that uses observables to always resolves from cache data first and then resolves
-from the server.
+- [ReactRelayNetworkModern][]: A network layer that uses the middleware pattern to separate responsibilities like
+  retrying, logging, caching and auth.
 
-You can implement offline first apps using a custom network layer.
+- [timobetina's example][timobetina]: The simplest Observable network layer you can start with.
+
+<!-- TODO: More, @sibelius do you have some good examples? -->
```

Comment:
I added three examples, there is also Peril's which is the simplest it could be: https://github.com/danger/dashboard.peril.systems/blob/master/src/lib/createRelayEnvironment.ts

Can you think of more?